x/stablestake/types/share_denom.go

```diff
@@ -1,5 +1,27 @@
 package types
 
-func GetShareDenom() string {
-	return "stablestake/share"
+import (
+	fmt "fmt"
+	"strconv"
+	"strings"
+)
+
+func GetShareDenomForPool(poolId uint64) string {
+	if poolId == PoolId {
+		return "stablestake/share"
+	}
+	return "stablestake/share/pool/" + strconv.FormatUint(poolId, 10)
+}
+
+// GetPoolIDFromPath retrieves the poolid from the given path in the format "stablestake/share/pool/poolid".
+func GetPoolIDFromPath(path string) (uint64, error) {
+	parts := strings.Split(path, "/")
+	if len(parts) != 4 || parts[0] != "stablestake" || parts[1] != "share" || parts[2] != "pool" {
```

Comment:
need to add case for usdc pool

Reply:
this function is used in tier module, USDC case is handled there.
.Denom == "stablestake/share"

Reply:
refactored